frigate/detectors/plugins/openvino.py

```diff
@@ -17,50 +17,119 @@ class OvDetectorConfig(BaseDetectorConfig):
-                break
-            detections[i] = [
-                object_detected[1],  # Label ID
-                float(object_detected[2]),  # Confidence
-                object_detected[4],  # y_min
-                object_detected[3],  # x_min
-                object_detected[6],  # y_max
-                object_detected[5],  # x_max
-            ]
-            i += 1
-
-        return detections
+            detections = np.zeros((20, 6), np.float32)
+            i = 0
+            for object_detected in results.data[0, 0, :]:
+                if object_detected[0] != -1:
+                    logger.debug(object_detected)
+                if object_detected[2] < 0.1 or i == 20:
+                    break
+                detections[i] = [
+                    object_detected[1],  # Label ID
+                    float(object_detected[2]),  # Confidence
+                    object_detected[4],  # y_min
+                    object_detected[3],  # x_min
+                    object_detected[6],  # y_max
+                    object_detected[5],  # x_max
+                ]
+                i += 1
+            return detections
+
+        elif(self.ov_model_type == ModelTypeEnum.yolox):
+            out_tensor = infer_request.get_output_tensor()
+            results = out_tensor.data
+            results[..., :2] = (results[..., :2] + self.grids) * self.expanded_strides
+            results[..., 2:4] = np.exp(results[..., 2:4]) * self.expanded_strides
+            image_pred = results[0, ...]
+
+            class_conf = np.max(image_pred[:, 5:5+self.num_classes], axis=1, keepdims=True)
+            class_pred = np.argmax(image_pred[: , 5:5+self.num_classes], axis=1)
+            class_pred = np.expand_dims(class_pred, axis=1)
+
+            conf_mask = (image_pred[:, 4] * class_conf.squeeze() >= 0.3).squeeze()
+            # Detections ordered as (x1, y1, x2, y2, obj_conf, class_conf, class_pred)
+            dets = np.concatenate((image_pred[:, :5], class_conf, class_pred), axis=1)
+            dets = dets[conf_mask]
+
+            ordered = dets[dets[:, 5].argsort()[::-1]][:20]
+
+            detections = np.zeros((20, 6), np.float32)
+            i = 0
+
+            for object_detected in ordered:
+                if i < 20:
+                # [x, y, h, w, box_score, class_no_1, ..., class_no_80],
+                    detections[i] = [
+                        object_detected[6],  # Label ID
+                        object_detected[5],  # Confidence
+                        (object_detected[1]-(object_detected[3]/2))/self.h,  # y_min
+                        (object_detected[0]-(object_detected[2]/2))/self.w,  # x_min
+                        (object_detected[1]+(object_detected[3]/2))/self.h,  # y_max
+                        (object_detected[0]+(object_detected[2]/2))/self.w,  # x_max
+                    ]
+                    #logger.info(object_detected)
+                    #logger.info(detections[i])
```

Comment:
Clean up commented code.  Remove or uncomment and log to debug instead?  I know it can sometimes be helpful to see the raw detections before they are filtered.

Reply:
Agreed. Done.